app/routes/datasets.js

```diff
@@ -16,21 +16,36 @@ export default class extends Route {
   model(params) {
     let dataset = this.modelFor('application').findBy('id', params.dataset_id);
     let yearcolumn = dataset.get('yearcolumn');
+    const prqlSchema = dataset.get('schemaname') === 'tabular' ? 'tabular' : 'mapc';
+
+    let token;
+    switch (dataset.get('db_name')) {
+      case 'ds':
+        token = config.database.dsToken
+        break;
+      case 'gisdata':
+        token = config.database.gisdataToken
+        break;
+      case 'towndata':
+        token = config.database.towndataToken
+        break;
+      default:
+        token = config.database.dsToken
+    }
```

Comment:
Instead of the classic `switch`, a more concise JS way of doing a conditional assignment would be

```js
const token = config.database[`${datasets.get('db_name')}Token`] || config.database.dsToken;
```

Reply:
You don't have to implement this, just throwing the pattern out there.